### PR TITLE
New nav bar

### PR DESCRIPTION
**Pull request by asharrahman02**
This the new nav bar, The mobile nav bar is still coming

### CODE REVIEW

File: src/components/Authentication/Utils/AuthUtils.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import {NavigateFunction} from "react-router-dom";
+import {signOut} from "firebase/auth";
+import {auth} from "../../../index";
+
+export async function signOutUser(navigate : NavigateFunction) {
+    await signOut(auth).then(value => {return Promise.resolve("User Signed Out")}).catch(reason => {return Promise.reject("User Not Signed Out")});
```
Comment (GiessC) [inline src/components/Authentication/Utils/AuthUtils.tsx]: do not need to surround return strings with Promise.resolve/reject

File: src/components/Dashboard/Dashboard.tsx [hunk 2]
```diff
@@ -69,13 +69,13 @@ export function Dashboard() {
...
-                <style> @import url('https://fonts.googleapis.com/css2?family=Pavanam&display=swap');</style>
-                <style> @import url('https://fonts.googleapis.com/css2?family=Inter:wght@300;400;700&display=swap');
-                </style>
-                <link rel="stylesheet"
-                      href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@48,400,0,0"/>
+                {/*<style> @import url('https://fonts.googleapis.com/css2?family=Pavanam&display=swap');</style>*/}
+                {/*<style> @import url('https://fonts.googleapis.com/css2?family=Inter:wght@300;400;700&display=swap');*/}
+                {/*</style>*/}
+                {/*<link rel="stylesheet"*/}
+                {/*      href="https://fonts.googleapis.com/css2?family=Material+Symbols+Outlined:opsz,wght,FILL,GRAD@48,400,0,0"/>*/}
```
Comment (GiessC) [inline src/components/Dashboard/Dashboard.tsx]: just delete this man

File: src/components/Nav/Sidebar/Sidebar.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    const currentRelPath = window.location.pathname;
+
+    return (
+        <div>
+            <AuthBar isOpen={isOpen} username={auth.currentUser?.displayName || ''}/>
+            <div className={`${styles.navbarContainer} ${isOpen ? styles.navbarExtended : ''}`} onMouseEnter={() => {setIsOpen(true)}} onMouseLeave={() => {if(!manuallyLocked){setIsOpen(false)}}}>
+                <h1 className={styles.navbarLogo}>S{isOpen ? 'heska' : ''}</h1>
+                <div className={styles.navigationGroup}>
+                    <NavButton icon={'home'} text={'Dashboard'} location={'/dashboard'} selected={'/dashboard' === currentRelPath} isSidebarOpen={isOpen}/>
```
Comment (GiessC) [inline src/components/Nav/Sidebar/Sidebar.tsx]: Here :
```
icon={IconComponentName}
```
In NavButton:
```
<icon />
```

File: src/components/Nav/Sidebar/Sidebar.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+                    <NavButton icon={'list'} text={'Sheska List'} location={'/sheskalist'} selected={'/sheskalist' === currentRelPath} isSidebarOpen={isOpen}/>
+                </div>
+                <div className={styles.settingGroup}>
+                    <NavButton icon={'settings'} text={"Account"} location={undefined} selected={false} isSidebarOpen={isOpen}/>
+                    <div onClick={() => {signOutUser(props.navigateFunction).then(r => {console.log('signed out user!')}).catch(reason => {console.log('failed to sign out user')})}}>
```
Comment (GiessC) [inline src/components/Nav/Sidebar/Sidebar.tsx]: just make this longer man stop doing the one line lambdas

File: src/components/Utils/CardUtil.tsx [hunk 5]
```diff
@@ -87,21 +89,21 @@ export async function getCardDescription (cardID: string): Promise<string> {
 export async function deleteCardImages (cardID: string) {
     const pathReference = ref(storage, '/users/'+ auth.currentUser?.uid.toString() + "/" + cardID + "/");
```
Comment (GiessC) [inline src/components/Utils/CardUtil.tsx]: convert to template string

File: src/components/Utils/CardUtil.tsx [hunk 6]
```diff
@@ -87,21 +89,21 @@ export async function getCardDescription (cardID: string): Promise<string> {
...
+                        console.log(error)
+                    })
+            });
+        }).catch((error) => {
         console.error(error);
     });
 }
 
 export async function deleteCard(cardID: string) {
     await deleteDoc(doc(db, 'users/' + auth.currentUser?.uid.toString() + "/sheska_list", cardID))
```
Comment (GiessC) [inline src/components/Utils/CardUtil.tsx]: convert to template string

File: src/components/Utils/DisplayFormikState.tsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+export const DisplayFormikState = (props: any) =>
+  <div style={{ margin: '1rem 0' }}>
+    <h3 style={{ fontFamily: 'monospace' }} />
```
Comment (GiessC) [inline src/components/Utils/DisplayFormikState.tsx:5]: delete this

File: src/components/Utils/SheskaCardDef.tsx [hunk 8]
```diff
@@ -2,35 +2,90 @@ import React from "react";
...
+
+    constructor(cardID: string, title: string, subtitle: string, description?: string, imageOrder?: string[], expectedAverage? : string[],
+                goal?: string[], amountRaised?: string[], guestsAbsorbFees?: boolean, dateCreated?: string, dateUpdated?: string) {
         this._cardID = cardID;
         this._title = title;
         this._subtitle = subtitle;
         if(description) {
             this._description = description;
+        } else {
+            this._description = '';
```
Comment (GiessC) [inline src/components/Utils/SheskaCardDef.tsx]: `this._description = description || '';`

Comment (GiessC) [reply src/components/Utils/SheskaCardDef.tsx]: use undefined instead of ''?

File: src/components/Authentication/Utils/AuthUtils.tsx (hunk 1 above)
Comment (asharrahman02) [reply src/components/Authentication/Utils/AuthUtils.tsx]: done

File: src/components/Dashboard/Dashboard.tsx (hunk 2 above)
Comment (asharrahman02) [reply src/components/Dashboard/Dashboard.tsx]: done

File: src/components/Nav/Sidebar/Sidebar.tsx (hunk 3 above)
Comment (asharrahman02) [reply src/components/Nav/Sidebar/Sidebar.tsx]: Done

File: src/components/Nav/Sidebar/Sidebar.tsx (hunk 4 above)
Comment (asharrahman02) [reply src/components/Nav/Sidebar/Sidebar.tsx]: Done

File: src/components/Utils/CardUtil.tsx (hunk 5 above)
Comment (asharrahman02) [reply src/components/Utils/CardUtil.tsx]: fixed I think

File: src/components/Utils/CardUtil.tsx (hunk 6 above)
Comment (asharrahman02) [reply src/components/Utils/CardUtil.tsx]: Done I think

File: src/components/Utils/DisplayFormikState.tsx (hunk 7 above)
Comment (asharrahman02) [reply src/components/Utils/DisplayFormikState.tsx:5]: Useful Util

File: src/components/Utils/SheskaCardDef.tsx (hunk 8 above)
Comment (asharrahman02) [reply src/components/Utils/SheskaCardDef.tsx]: fixed

**Review by GiessC — APPROVED** [review]

**Pull request merged**